dataservice/api/common/validation.py

```diff
@@ -97,3 +97,39 @@ def validate_list(input_items):
                 f'valid set: {", ".join(valid_items)}'
             )
     return validate_list
+
+
+def alphanumeric_str_validation_generator(field_name, quantifier_regex='+'):
+    """
+    Generate a function to validate alphanumeric strings.
+
+    Length of string is not validated by default, but can be using
+    `quantifier_regex` which must be a valid quantifier regex pattern.
+    For example quantifier_regex={3, 6} will check that string has a min
+    length of 3 and max length of 6.
+
+    :param field_name: Name of the field/attribute that is being validated
+    :type field_name: str
+    :param quantifier_regex: Regex pattern to check for min, max length.
+    :type quantifier_regex: str
+
+    :returns: validation function
+    """
+    assert isinstance(quantifier_regex, str), (
```

Comment:
Maybe you'd be safer asserting that this is valid regex via `re.compile` or similar?

Reply:
Good idea 👍